pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -317,11 +324,19 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 			}
 			log.Info("machine has current version", "name", machine.GetName(),
 				"version", node.Annotations[nodeconfig.VersionAnnotation])
-			// version annotation exists with a valid value, node is fully configured, do nothing.
+			// version annotation exists with a valid value, node is fully configured.
+			// configure prometheus if it is not configured
+			if err := pc.Configure(); err != nil {
```

Comment:
I am wondering why we make a call to `pc.Configure()` regardless of the machine's status. For my learning, could you please explain why we'd want to configure Prometheus again if the machine is already `Running` for example? I am assuming here that a machine would already have Prometheus configured if it has reached the `Running` state.

Reply:
This call is required when a WMCO pod restarts. It creates a [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) which causes Endpoints object to revert all the changes.

We are creating that service [here](https://github.com/openshift/windows-machine-config-operator/pull/142/files#diff-ccf7cfc3caf950d956bbe879bfbd3fd978235b1e00781499c93510d8e95ff994R81). Hence we need to reconfigure

Reply:
I almost feel like we need a separate node controller for this, as this doesn't deal with the machine object at all, and isnt fitting in well within this controller. I think a node controller makes more sense, and will help clean up the code, as its getting somewhat convoluted in my opinion.